test/fast/FastHistoryFacet.test.ts

```diff
@@ -101,13 +101,24 @@ describe('FastHistoryFacet', () => {
         .revertedWith(INTERNAL_METHOD);
     });
 
-    it('adds an entry to the supply proof list', async () => {
-      // TODO: We need to find a way to be able to call this functions.
-      // await history.burnt(3, 'Three');
-      // const [[{ amount, ref, blockNumber }],] = await history.paginateSupplyProofs(0, 1);
-      // expect(amount).to.eq(3);
-      // expect(ref).to.eq('Three');
-      // expect(blockNumber.toNumber()).to.be.greaterThan(1);
+    describe('as the diamond', async () => {
+      let tokenAsItself: FastHistoryFacet;
+
+      beforeEach(async () => {
+        tokenAsItself = await impersonateDiamond(history);
```

Comment:
I've gone for the basic option at the moment which means there needs to be the `stopImpersonatingAccount` called in the `afterEach`. Maybe later we can tidy this.

Reply:
It looks good for now.